src/record.rs

```diff
@@ -230,6 +279,11 @@ impl Header {
         bincode::serialized_size(&self).expect("calc record serialized size")
     }
 
+    pub(crate) fn set_offset_checksum(&mut self, blob_offset: u64, header_checksum: u32) {
+        self.blob_offset = blob_offset;
+        self.header_checksum = header_checksum;
```

Comment:
Can you add `debug_assert` to verify, that new checksum equals to `self.crc32()`